indico/modules/events/timetable/client/js/actions.ts

```diff
@@ -304,3 +302,59 @@ export function closeModal() {
 export function setCurrentDate(date: Moment, eventId: number) {
   return {type: SET_CURRENT_DATE, date, eventId};
 }
+
+/**
+ * Manages a queue of requests to be processed in order.
+ * This ensures all timetable requests are handled sequentially.
+ *
+ * TODO(tomas): On error, we just call `handleAxiosError` for now.
+ * In the future, we might consider some undo mechanism.
+ */
+class RequestQueue {
+  requests: (() => Promise<object>)[];
+  working: boolean;
+
+  constructor() {
+    this.requests = [];
+    this.working = false;
+  }
+
+  submitRequest(request) {
+    this.requests.push(request);
+    if (!this.working) {
+      this._processRequests().catch(error => {
+        console.error('Error processing request queue:', error);
+      });
+    }
+  }
+
+  async _processRequests() {
+    this.working = true;
+    while (this.requests.length > 0) {
```

Comment:
Nitpicky but this doesn't need the number comparison. Not sure if it's for readability

```suggestion
    while (this.requests.length) {
```